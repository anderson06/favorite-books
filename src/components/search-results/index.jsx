import SearchResults from './SearchResults';
import Pagination from './Pagination';

export { SearchResults, Pagination };
