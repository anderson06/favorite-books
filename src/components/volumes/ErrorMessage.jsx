import React from 'react';

const Loading = () => (
  <div className="fb-fail-message">
    <h5>Ocorreu um erro</h5>
    <p>Tente novamente mais tarde</p>
  </div>
);

export default Loading;

