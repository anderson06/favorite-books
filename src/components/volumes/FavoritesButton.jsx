import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const FavoritesButton = ({ favorite }) => {
  return (
    <a
      href="#!"
      className={classNames(
        'secondary-content',
        'fb-favorite',
        { 'fb-favorite--on': favorite }
      )}
    >
      <i className="material-icons">grade</i>
    </a>
  );
};

FavoritesButton.propTypes = {
  favorite: PropTypes.bool,
};

FavoritesButton.defaultProps = {
  favorite: false,
};

export default FavoritesButton;

