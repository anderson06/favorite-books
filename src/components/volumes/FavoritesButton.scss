.fb-favorite {
  color: #9e9e9e;

  &--on {
    color: #ff9800;
  }
}

