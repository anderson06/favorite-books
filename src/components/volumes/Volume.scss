.fb-volume {
  cursor: pointer;
}

.fb-volume-title {
  margin-right: 20px;
}

.fb-thumbnail-icon {
  line-height: 42px;
}

.highlight {
  font-weight: bold;
}

.fb-volume-cover {
  position: absolute;
  width: 42px;
  overflow: hidden;
  left: 15px;
  display: inline-block;
  vertical-align: middle;
  border: 0;
}
