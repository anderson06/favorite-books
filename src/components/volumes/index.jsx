import VolumeList from './VolumeList';
import Volume from './Volume';
import VolumeInfo from './VolumeInfo';
import FavoritesButton from './FavoritesButton';
import Loading from './Loading';
import ErrorMessage from './ErrorMessage';

export { VolumeList, Volume, FavoritesButton, VolumeInfo, Loading, ErrorMessage };

