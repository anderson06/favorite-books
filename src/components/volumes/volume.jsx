import React from 'react';
import PropTypes from 'prop-types';
import './Volume.scss';

const Volume = ({ thumbnail, title, description }) => {
  let image = (
    <div className="circle valign-wrapper center-align teal lighten-2">
      <i className="material-icons fb-thumbnail-icon">library_books</i>
    </div>
  );
  if (thumbnail) {
    image = <img src={thumbnail} alt="book cover" className="circle" />;
  }
  return (
    <li
      className="fb-volume collection-item avatar"
    >
      {image}
      <span className="title truncate">{title}</span>
      <p className="fb-truncate">{description}</p>
      <a href="#!" className="secondary-content">
        <i className="material-icons">grade</i>
      </a>
    </li>
  );
};

Volume.propTypes = {
  description: PropTypes.string,
  thumbnail: PropTypes.string,
  title: PropTypes.string,
};

Volume.defaultProps = {
  description: '',
  thumbnail: '',
  title: '',
};

export default Volume;
