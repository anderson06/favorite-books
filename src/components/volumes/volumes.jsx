import React from 'react';
import PropTypes from 'prop-types';
import Volume from './volume';

const Volumes = ({ volumes }) => {
  if (volumes.length === 0) {
    return null;
  }
  const listVolumes = volumes.map(volume => (
    <Volume key={volume.id} {...volume} />
  ));
  return (
    <ul className="fb-volumes collection">
      {listVolumes}
    </ul>
  );
};

Volumes.propTypes = {
  volumes: PropTypes.arrayOf(PropTypes.shape(Volume.propTypes)),
};

Volumes.defaultProps = {
  volumes: [],
};

export default Volumes;

