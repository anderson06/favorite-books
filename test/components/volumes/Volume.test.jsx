import React from 'react';
import { shallow } from 'enzyme';
import { Volume, FavoritesButton } from '../../../src/components/volumes';
import { aVolume } from '../../builders/volume';

describe('<Volume />', () => {
  it('renders a book placeholder image when no thumbnail is passed', () => {
    const volume = aVolume();
    delete volume.thumbnail;
    const wrapper = shallow(<Volume {...volume} />);
    expect(wrapper.find('.placeholder').length).toBe(1);
  });

  it('should set onClick prop to FavoritesButton', () => {
    const onFavoriteButtonClick = () => {};
    const wrapper = shallow(
      <Volume
        {...aVolume()}
        onFavoriteButtonClick={onFavoriteButtonClick}
      />);
    const { onClick } = wrapper.find(FavoritesButton).props();
    expect(onClick).toBe(onFavoriteButtonClick);
  });

  it('should set favorite prop to FavoritesButton', () => {
    const wrapper = shallow(<Volume {...aVolume()} favorite />);
    const { favorite } = wrapper.find(FavoritesButton).props();
    expect(favorite).toBe(true);
  });

  it('should set volumeId prop to FavoritesButton', () => {
    const volume = aVolume();
    const wrapper = shallow(<Volume {...volume} />);
    const { volumeId } = wrapper.find(FavoritesButton).props();
    expect(volumeId).toBe(volume.id);
  });
});
