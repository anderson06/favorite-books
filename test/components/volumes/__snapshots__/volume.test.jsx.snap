// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Volume /> renders a book placeholder image when no thumbnail is passed 1`] = `
<li
  className="fb-volume collection-item avatar"
>
  <div
    className="circle valign-wrapper center-align teal lighten-2"
  >
    <i
      className="material-icons fb-thumbnail-icon"
    >
      library_books
    </i>
  </div>
  <span
    className="title truncate"
  >
    Fake title
  </span>
  <p
    className="fb-truncate"
  >
    Fake description
  </p>
  <a
    className="secondary-content"
    href="#!"
  >
    <i
      className="material-icons"
    >
      grade
    </i>
  </a>
</li>
`;

exports[`<Volume /> renders correctly 1`] = `
<li
  className="fb-volume collection-item avatar"
>
  <img
    className="circle"
    src="fake-thumbnail.png"
  />
  <span
    className="title truncate"
  >
    Fake title
  </span>
  <p
    className="fb-truncate"
  >
    Fake description
  </p>
  <a
    className="secondary-content"
    href="#!"
  >
    <i
      className="material-icons"
    >
      grade
    </i>
  </a>
</li>
`;
